File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.java
class Solution {
    public int maxAdjacentDistance(int[] nums) {
        int n = nums.length;
        int max_element = Math.abs(nums[0] - nums[n-1]);
        for(int i = 0; i < n-1; i++){
            max_element = Math.max(max_element, Math.abs(nums[i] - nums[i+1]));
        }
        return max_element;
    }
}

File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.py
class Solution:
    def maxAdjacentDistance(self, nums: List[int]) -> int:
        max_element = abs(nums[0] - nums[-1])
        for i in range(len(nums)):    
            diff = abs(nums[i] - nums[i-1])
            max_element = max(max_element, diff)
        return max_element

        